src/context/TreeContext.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from "react";
+
+interface TreeNode {
+  name: string;
+  props: Array<PropFormValue>;
+  state: Array<StateFormValue>;
+  parent: string;
+  children?: Array<TreeNode>;
+}
+
+// setTree optional, otherwise 'value' prop is missing a property
+type TreeContextState = {
+  tree: Array<TreeNode>;
+  setTree?: React.Dispatch<React.SetStateAction<Array<TreeNode>>>;
+};
+
+const treeContextDefault: TreeContextState = {
+  tree: [],
+  setTree: () => {},
+};
+
+const TreeContext = React.createContext<TreeContextState>(treeContextDefault);
+TreeContext.displayName = "TreeContext";
+
+export default function TreeContextProvider({ children }: AppProps) {
+  const [tree, setTree] = React.useState<TreeNode[]>([]);
+
+  const value = { tree };
+
+  return <TreeContext.Provider value={value}>{children}</TreeContext.Provider>;
+}
+
+export function useTreeContext() {
+  const tree = React.useContext(TreeContext);
+
+  if (!tree) {
```

Comment:
we have a `treeContextDefault` , so the context value won't be undefined.

Reply:
I see what you mean, but if the context hook is requested outside of its provider, won't it be undefined?

Reply:
No

Reply:
Okay I found a solution. Default value is undefined now, so this line will work.